WcaOnRails/spec/controllers/competitions_controller_spec.rb

```diff
@@ -660,4 +660,64 @@
       end
     end
   end
+
+  describe 'GET #edit_events' do
+    context 'when not signed in' do
+      sign_out
+
+      it 'redirects to the sign in page' do
+        get :edit_events, id: competition.id
+        expect(response).to redirect_to new_user_session_path
+      end
+    end
+
+    context 'when signed in as an admin' do
+      sign_in { FactoryGirl.create :admin }
+
+      it 'shows the edit competition events form' do
+        get :edit_events, id: competition.id
+        expect(response).to render_template :edit_events
+      end
+    end
+
+    context 'when signed in as a regular user' do
+      sign_in { FactoryGirl.create :user }
+
+      it 'does not allow access' do
+        expect {
+          get :edit_events, id: competition.id
+        }.to raise_error(ActionController::RoutingError)
```

Comment:
`  before_action -> { redirect_unless_user(:can_manage_competition?, competition_from_params) }, only: [:edit, :update, :edit_events, :update_events]`

This calls `competition_from_params`

`  private def competition_from_params
    competition = Competition.find(params[:competition_id] || params[:id])
    if !competition.user_can_view?(current_user)
      raise ActionController::RoutingError.new('Not Found')
    end
    competition
  end`

which gives a routing error is the user can't view the competition. Do you think this is fine or should I change the way this is being processed?